### ISSUE TITLE
Exception trying to send an exception response

### ISSUE THREAD

**Issue by galvesribeiro** [dotnet/orleans#2600]
In 2.0 TP, whenever an exception happens and must be serialized to send to the remote caller this exception happen:

```

Value cannot be null.
Parameter name: meth
Exception = System.ArgumentNullException: Value cannot be null.
Parameter name: meth
   at System.Reflection.Emit.DynamicILGenerator.Emit(OpCode opcode, MethodInfo meth)
   at Orleans.Runtime.InsideRuntimeClient.CreateExceptionPrepForRemotingMethod()
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at Orleans.Runtime.InsideRuntimeClient.PrepareForRemoting(Exception exception)
   at Orleans.Runtime.InsideRuntimeClient.SafeSendExceptionResponse(Message message, Exception ex)
```
And
```
[2017-01-16 21:27:44.610 GMT    11      WARNING 101532  InsideRuntimeClient     192.168.1.80:22222] Exception trying to send an exception response: Value cannot be null.
```

Reuben found out that this is the offending code:

```
var methodInfo = typeof(Exception).GetMethod(
                "PrepForRemoting",
                BindingFlags.Instance | BindingFlags.NonPublic);
```

Looks like `Exception.PrepForRemoting` is not available in .Net Core.

cc: @ReubenBond

**Comment by ReubenBond** [comment]
Note that this is on .NET Core. Does this happen using 2.0TP on Full .NET, too, @galvesribeiro?

**Comment by galvesribeiro** [comment]
Just confirmed, full framework has it, .Net Core doesn't.

**Comment by galvesribeiro** [comment]
Implemented on #2601
